synapse/visibility.py

```diff
@@ -207,15 +207,12 @@ async def filter_event_for_clients_with_state(
             ):
                 allowed_user_ids.discard(user_id)
 
-    if event.internal_metadata.outlier:
-        # Normally these can't be seen by clients, but we make an exception for
-        # for out-of-band membership events (eg, incoming invites, or rejections of
```

Comment:
For reference, out-of-band membership events has some docs written here: https://github.com/matrix-org/synapse/blob/develop/docs/development/room-dag-concepts.md#out-of-band-membership-events

We might want to better clarify it once we figure out what we want/can do